### ISSUE TITLE
Implement In-memory IDataStore for unit testing

### ISSUE THREAD

**Issue by Sewer56** [Nexus-Mods/NexusMods.App#283]
# Currently

We have an `IDataStore` (database) that is backed by SQLite3. This component is needed for some unit tests. 

However, using the SQLite-backed `IDataStore` for unit testing presents a problem, as it requires we write data to disk. This can slow down the tests and risk leaving junk files behind if the user either stops debugging during a debugging session [which kills application] or does not properly clean up files from their test code. 

Currently, writing completely in-memory unit tests in some cases is not possible because we lack an implementation of an `IDataStore` that stores all of its information in memory.

# We want

We want to create an in-memory `IDataStore` implementation to to enable for cleaner unit testing. This new implementation should store all its information in memory and not write any files to disk, making it more suitable for unit testing purposes.

# Supporting Information
N/A

# Design
The design of this improvement should include:

1. A new class that implements the `IDataStore` interface (e.g. `InMemoryDataStore`), using an in-memory data structure (such as nested Dictionary) to store data instead of writing it to disk.

## Further Work

Convert existing tests relying on `SqliteDataStore` to use `InMemoryDataStore`. Open another issue for this once this is complete.

**Comment by Al12rs** [comment]
As an alternative, from my knowledge SQLite3 can be configured to work in memory, which should offer the performance improvements and avoid the disk writes. This would need to be better investigated to be sure.

One advantage with this approach could be that the tests can also cover any regressions which are related to the SQLite3 database interactions.

Don't know if simply changing it to in memory satisfies the requirements here or if the intentions where to obtain an even leaner and simpler way to test

**Comment by halgari** [comment]
Yeah, the easiest way to do this would probably to use a in-memory instance
